std/traits.d

```diff
@@ -3550,6 +3551,176 @@ enum hasMember(T, string name) = __traits(hasMember, T, name);
 }
 
 /**
+ * Whether the symbol represented by the string, member, is a static member of T.
+ *
+ * Params:
+ *     T = Type containing symbol $(D member).
+ *     member = Name of symbol to test that resides in $(D T).
+ *
+ * Returns:
+ *     $(D true) iff $(D member) is static.
+ */
+template isStaticMember(T, string member)
+    if (__traits(hasMember, T, member))
+{
+    import std.meta : Alias;
+    alias sym = Alias!(__traits(getMember, T, member));
```

Comment:
I don't think it causes that big of a problem that it's worth waiting for a DIP that's still being written. When or even if that DIP is introduced, it won't cause any breakage to existing code and it'll be a small patch to allow private protected symbols for isStaticMember. I don't see there being very many cases where it would need to be made known whether a private member is static in another module that it won't be able to access anyways.